### PR TITLE
Issue 34 - Failed creation of a submission returns to form

### PR DESCRIPTION
**Pull request by flyinggrizzly**
#### What does this PR do?

Fixes [issue 34](https://github.com/nodunayo/speakerline/issues/34), where invalid create actions on the SubmissionsController would redirect to the speakers index, instead of back to the form.

Currently, when a user tries to create an invalid submission (submission of a proposal, not a form submission), the app redirects them to the speakers index instead of back to the form.

This PR changes this flow so the user is redirected back to the submission creation form.

It also changes the spec for the `SubmissionsController` so that it tests not submissions with all invalid attributes, but submissions with all attributes invalid except for the proposal ID.

This PR does *not* test the case when all submitted attributes are invalid, because this actually opens questions about the `SubmissionsController` routing behavior: it displays Rails errors when the user accesses `/submissions/new` without a proposal ID URL attribute. ([Issue 48](https://github.com/nodunayo/speakerline/issues/48) opened for this).

##### Why was this work done? Is there a related Issue?

Fixes [issue 34](https://github.com/nodunayo/speakerline/issues/34); also part of 24 Pull Requests.

#### Where should a reviewer start?

The only changes are in the `SubmissionsController` and its spec file. 

Worth noting, in the spec, the `let(:proposal)...` declaration was moved so it can be accessed in both contexts, and the invalid attrs context no longer uses an anonymous double because it needs a valid proposal attribute.

The additional expectation that the action `should retain its proposal id` (line 37) is technically redundant because the redirect expectation on line 36 already catches this, but feels useful in case something else happens to the URL structure.

#### Are there any manual testing steps?

1. (Add a speaker)
2. (Add a proposal for the speaker)
3. Create a submission for the proposal without selecting a "Accepted/Rejected" result
4. Confirm you are redirected back to the submission form

---

### CODE REVIEW

**Comment by nodunayo** [comment]
Great! Love this write-up. Thanks @flyinggrizzly. Will check this out now! :-)

File: spec/controllers/submissions_controller_spec.rb [hunk 1]
```diff
@@ -20,18 +22,21 @@
       it { should redirect_to(proposal_path(assigns(:submission).proposal)) }
     end
 
-    context 'with invalid attributes' do
-      let(:invalid_submission) { double }
+    context 'with a valid submission attribute and invalid event and proposal id attributes' do
```
Comment (nodunayo) [inline spec/controllers/submissions_controller_spec.rb]: Hi @flyinggrizzly — a bit confused about the wording here.

What does the `valid submission attribute` refer to? And isn't the proposal id valid here?

Comment (flyinggrizzly) [reply spec/controllers/submissions_controller_spec.rb]: Shoot. Yea, got myself mixed up. That should be `context 'with valid proposal attribute and invalid event and result attributes'`

I'll fix this and push again.

**Comment by flyinggrizzly** [comment]
@nodunayo fixed the attribute mixup; let me know if it needs any more changes!

**Comment by nodunayo** [comment]
@flyinggrizzly Just curious...when you made the change in the test suite, did you rebase locally (since it's still one commit) and then re-push? (Curious about how that flow works for PRs).

**Comment by nodunayo** [comment]
Yay, shipped! 👍

**Comment by flyinggrizzly** [comment]
🎉

@nodunayo yea I rebased on my fork—didn’t seem worth giving the goof its own commit

**Pull request merged**